orchagent/portsorch.cpp

```diff
@@ -2553,6 +2581,221 @@ bool PortsOrch::removeBridgePort(Port &port)
     return true;
 }
 
+bool PortsOrch::setHostVlanAdminState(int vlan_id, const string &admin_status)
+{
+    SWSS_LOG_ENTER();
+
+    /* The command should be generated as:
+ *      /sbin/ip link set Vlan{{vlan_id}} {{admin_status}}
+ *           */
+    const std::string cmds = std::string("")
+      + IP_CMD + " link set " + VLAN_PREFIX + std::to_string(vlan_id) + " " + admin_status;
+
+    std::string res;
+    EXEC_WITH_ERROR_THROW(cmds, res);
+
+    return true;
+}
+
+string PortsOrch::getPortOperStatus(string port_alias)
+{
+    string port_oper_status = "down";
+    vector<FieldValueTuple> port_values;
+    /*  update admin status */
+    if (!m_appPortTable.get(port_alias, port_values))
+    {
+        SWSS_LOG_ERROR("failed read appPortTable, key: %s", port_alias.c_str());
+        return false;
+    }
+    for (FieldValueTuple i : port_values)
+    {
+        if (fvField(i) == "oper_status")
+        {
+            port_oper_status = fvValue(i);
+            break;
+        }
+    }
+    return port_oper_status;
+}
+
+string PortsOrch::updateLagStatus(Port &lag)
```

Comment:
why this function is called 'updateLagStatus'?

Reply:
member port of vlan may be type of Lag，this function update status of lag port by check all its memer port's oper_status